app/helpers/pagelets_helper.rb

```diff
@@ -67,8 +67,70 @@ def th_content(col)
     sort col[:key], as: col[:label], default: col[:default_sort] || 'ASC'
   end
 
+  def mount_column_selector
+    url = "#{api_users_path}/#{User.current[:id]}/table_preferences"
+    has_preference = !@selected_columns.nil?
+    columns = columns_view
+    react_component("ColumnSelector", data:
+      {
+        url: url,
+        controller: controller_name,
+        columns: columns,
+        initialColumns: columns,
+        hasPreference: has_preference,
```

Comment:
We can skip allocation for the variable if it's not used anywhere else.
```suggestion
        hasPreference: @selected_columns.present?,
```